azure/scope/machinepool.go

```diff
@@ -958,9 +958,9 @@ func (m *MachinePoolScope) HasReplicasExternallyManaged(_ context.Context) bool
 }
 
 // ReconcileReplicas ensures MachinePool replicas match VMSS capacity if replicas are externally managed by an autoscaler.
-func (m *MachinePoolScope) ReconcileReplicas(ctx context.Context, vmss *azure.VMSS) error {
```

Comment:
This change to the interface could be up for discussion, but it would have to ship with a minor release, we wouldn't be able to include this as a bug fix in a patch release. I assume we want to get this fix out, so I'd advise removing this change from this PR.